### DISCUSSION TITLE
Is it possible to modify from where it displays the balances the account modal?

### DISCUSSION THREAD

**Discussion by guillermoriv** [rainbow-me/rainbowkit#1421] [Q&A] [answered]
I am using an smart wallet, so I can executeTransaction's and also relay them if I want, so I have a `<ConnectButton.Custom>`, is there any way to decide from where it fetches the balances and also the display name?, currently I have something like this, based on my hook that fetches the safeWallet:

```ts
{isSafeWallet ? safeAddress : account.displayName}
```
But I would also like to modify this as well:

![image](https://github.com/rainbow-me/rainbowkit/assets/61250854/cc76802e-822c-4256-8e9f-ae55badd7549)

**Comment by DanielSinclair** [comment]
@guillermoriv We tap into the `useEnsName` and `useBalance` Wagmi hooks for this functionality. Can you walk me through the smart contract wallet setup and display name/balance issues that you're seeing?

**Reply by guillermoriv** [reply]
Hey @DanielSinclair sorry for the late response, yes and thanks for the response, what we are trying to do is, when a user has a safe wallet with the wallet that he's connected with, we want to display all the information that we see on rainbowkit with this safe wallet instead of the wallet that he's connected with (EOA), obviously we want to keep the walletClient the same so we can still encodeData to this safe wallet and execute aggregated transactions.

I provided this context, so you have an understanding of what we are trying to do, let me know if this is possible, thanks!

I will provide as well some pictures of what we want to modify:

![image](https://github.com/rainbow-me/rainbowkit/assets/61250854/a47fa68d-08ce-4ab1-94a8-41980b17788d)
![image](https://github.com/rainbow-me/rainbowkit/assets/61250854/3cd05ef4-8e89-4171-a11f-74739334c8af)

In this situation we want to substitute the EOA address for the safe wallet address (without obviously modifiying the walletClient, so we can still execute transactions with the EOA), I know that I can archive showing the address thorugh the `<ConnectButton.Custom>` but I sill will lose functionality for copying the address and also showing recent transactions.

**Reply by DanielSinclair** [reply, marked as answer]
I see. This is currently not possible because RainbowKit relies on Wagmi's account state. Wagmi itself wasn't designed for the Smart Contract Wallet use case and only exposes the concept of 1 connected account. You will unfortunately have to do some heavy lifting on your end to smooth this out, and a custom Account Modal would be part of the equation. 

Do also consider that the user will require gas for smart contract wallet signatures, just like any other kind of transaction, so you may actually want to keep RainbowKit's exposure of the EOA and its balance, and instead reorient your dApp to make clear to the user that the smart contract wallet is a separate entity. 

I'd also recommend potentially directing users to interact with your dApp through the Safe Browser, which abstracts away some of these concerns and only exposes the Safe address to the dApp, and handles signature collection from the EOA in its own UI.
